x-pack/plugins/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from 'kibana/server';
+// import uuid from 'uuid/v4';
+// import { Job as AnomalyDetectionJob } from '../../../../ml/server';
+import { PromiseReturnType } from '../../../../observability/typings/common';
+import { Setup } from '../helpers/setup_request';
+
+export type CreateAnomalyDetectionJobsAPIResponse = PromiseReturnType<
+  typeof createAnomalyDetectionJobs
+>;
+export async function createAnomalyDetectionJobs(
+  setup: Setup,
+  environments: string[],
+  logger: Logger
+) {
+  const { ml } = setup;
+  if (!ml) {
+    return [];
+  }
+  const mlCapabilities = await ml.mlSystem.mlCapabilities();
+  if (
+    !(
+      mlCapabilities.mlFeatureEnabledInSpace &&
+      mlCapabilities.isPlatinumOrTrialLicense
+    )
+  ) {
+    logger.warn('Anomaly detection integration is not availble for this user.');
+    return [];
+  }
+  logger.info(
+    `Creating ML anomaly detection jobs for environments: [${environments}]...`
+  );
+
+  const result = await Promise.all(
+    environments.map((environment) => {
+      return configureAnomalyDetectionJob({ ml, environment });
+    })
+  );
+  return result;
+}
+
+async function configureAnomalyDetectionJob({
+  ml,
+  environment,
+}: {
+  ml: Required<Setup>['ml'];
+  environment: string;
+}) {
+  const convertedEnvironment = environment; // TODO convert to ML id compatible format
+  const randomToken = '1234'; // TODO use generated UUID
+  const moduleId = 'apm_transaction';
+  const prefix = `apm-${convertedEnvironment}-${randomToken}-`;
+  const groups = ['apm', convertedEnvironment];
+  const indexPatternName = 'apm-*-transaction-*';
+  const query = {
+    bool: {
+      filter: [
+        { term: { 'processor.event': 'transaction' } },
+        { exists: { field: 'transaction.duration.us' } },
+        { term: { 'service.environment': environment } },
+      ],
+    },
+  };
+  const useDedicatedIndex = false;
+  const startDatafeed = true;
+  const start = Date.now(); // TODO what should these start/end values be?
+  const end = Date.now();
+  const jobOverrides = [
+    {
+      custom_settings: {
+        job_tags: {
+          'service.environment': environment,
+        },
+      },
+    },
+  ];
+  const datafeedOverrides: any[] = []; // TODO is this required?
+
+  return ml.modules.saveModuleItems(
```

Comment:
@jgowdyelastic with this many arguments it would be better with and object like:
```
return ml.modules.saveModuleItems({
  moduleId: 'my-module-id',
  prefix: 'apm-'
  // ...
})
```

A quick search revealed this is not used outside ml yet (and only used in a total of two files) so perhaps we can update this for 7.9?

Reply:
yes, I like this idea.
There is also some naming conflicts between `setupModuleItems`, `saveModuleItems` and the endpoint's name `setup`
I'll create a PR now which adds this suggested change and also renames the shared function to `setup`

Reply:
https://github.com/elastic/kibana/pull/70589

Reply:
Beautiful.